diff --git a/addtoany.module b/addtoany.module
index a601800..bd9275e 100644
--- a/addtoany.module
+++ b/addtoany.module
@@ -42,6 +42,19 @@ function addtoany_entity_extra_field_info() {
     );
   }
 
+  $moduleHandler = \Drupal::service('module_handler');
+  if ($moduleHandler->moduleExists('media_entity')){
+    /** @var Drupal\media_entity\MediaInterface $bundle */
+    foreach(Drupal\media_entity\Entity\Media::loadMultiple() as $bundle) {
+      $extra['media'][$bundle->bundle()]['display']['addtoany'] = [
+        'label' => t('AddToAny'),
+        'description' => t('Share buttons by AddToAny'),
+        'weight' => 5,
+        'visible' => FALSE,
+      ];
+    }
+  }
+
   return $extra;
 }
 
@@ -61,6 +74,21 @@ function addtoany_node_view(array &$build, EntityInterface $node, EntityViewDisp
 }
 
 /**
+ * Implements hook_ENTITY_TYPE_view().
+ */
+function addtoany_media_view(array &$build, EntityInterface $entity, EntityViewDisplayInterface $display, $view_mode) {
+  if ($display->getComponent('addtoany') && $entity->isPublished()) {
+    $build['addtoany'] = [
+      // Not using #markup because data attribute name gets mangled
+      // @see https://www.drupal.org/node/2609928
+      // '#markup' => addtoany_create_node_buttons($node),
+      '#addtoany_html' => addtoany_create_node_buttons($entity),
+      '#theme' => 'addtoany_standard',
+    ];
+  }
+}
+
+/**
  * Implements hook_page_attachments().
  */
 function addtoany_page_attachments(&$page) {
